rover-apps/science/src/AdafruitSTEMMAModule.cpp

```diff
@@ -0,0 +1,32 @@
+#include "AdafruitSTEMMAModule.h"
+
+#include <ratio>
+
+#include "Module.h"
+#include "mbed.h"
+
+AdafruitSTEMMAModule::AdafruitSTEMMAModule(PinName sda, PinName scl) : moistureSensor(sda, scl) {
+  static int numFails = 0;
+  bool resetStatus    = false;
+
+  while (!resetStatus && numFails < 10) {
+    resetStatus = moistureSensor.reset();
+  }
+  // if still error then log through printf (currently), plan to later log through LOGGER (TODO)
+  printf("Cannot Reset Moisture Sensor");
```

Comment:
This line will currently print regardless of whether the sensor reset failed or succeeded